src/robusta/core/sinks/slack/slack_sink.py

```diff
@@ -9,7 +12,141 @@ def __init__(self, sink_config: SlackSinkConfigWrapper, registry):
         super().__init__(sink_config.slack_sink, registry)
         self.slack_channel = sink_config.slack_sink.slack_channel
         self.api_key = sink_config.slack_sink.api_key
-        self.slack_sender = slack_module.SlackSender(self.api_key, self.account_id, self.cluster_name, self.signing_key)
+        self.slack_sender = slack_module.SlackSender(
+            self.api_key, self.account_id, self.cluster_name, self.signing_key, self.slack_channel
+        )
+
+    def write_finding(self, finding: Finding, platform_enabled: bool) -> None:
+        if self.grouping_enabled:
+            self.handle_notification_grouping(finding, platform_enabled)
+        else:
+            self.slack_sender.send_finding_to_slack(finding, self.params, platform_enabled)
```

Comment:
This is a more widespread problem that exists in many places in the codebase and its reason is that `self.params` is set in the constructor for `SinkBase`, and `SinkBase` knows nothing about the final type of params, so it has this value type-hinted as `SinkBaseParams`.

Moving the `params` initialization from `SinkBase` to the final sink params classes (like `SlackSink`) would add quite a bit of noise to this PR and result in a lot of code repetition.

Instead of that, I added a type hint (`params: SlackSinkParams`) to `SlackSink` so that it's clear what the actual type of this field is, both for readers and type checkers. What do you think about this solution?

We could probably solve it in some other, even more smart way, but I don't think this PR is about that, so maybe let's leave thinking about and fixing this issue in a perfect way for the future?